### DISCUSSION TITLE
Is there a manual way to overcome Directus migration error on SQLite?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#12796] [Archive]
For those using SQLite, we are stuck to 9.7
- [Sqlite migration directus v9.7.0 to v9.8.0 fails](https://github.com/directus/directus/issues/12529)
- [Sqlite migration directus v9.7.1 to v9.9.0 fails](https://github.com/directus/directus/issues/12795)

Same error on both 
`Error: DROP TABLE "directus_users" - SQLITE_CONSTRAINT: FOREIGN KEY constraint failed`

Is there another way to be able to benefit of the newest versions?
Something like modifying `directus_users` by hand to elevate it to the new schema, even through export/import?

**Comment by Pachat** [comment]
Because it looks like a Knex problem, i looked at that project for the first time.

Concerning sqlite, there is a thread  [switch to better-sqlite3](https://github.com/knex/knex/issues/5097) in which we read that it is a good idea for knex v.2 At present knex is at 1.0.7. It seems a long way to get to v.2
And i get feared when i see that they have 558 bugs open...

What to do if I get stuck to Directus v9.7.1?

Should I migrate? to MySQL or PostgreSQL?
And, in case I should, what is the way to migrate my database:
- install Directus 9.9.0 on a new PostgreSQL database and then migrate my tables? what about the users?
- stay with Directus 9.7.1, migrate all the tables of the SQLite database to a new PostgreSQL database and then update to Directus 9.9.0?

**Comment by rijkvanzanten** [comment]
You could theoretically prevent the migration from running by adding it's key to directus_migrations manually, and then apply the change the migration tries to make by hand, though I wouldn't recommend it. We're tracking the fix for that issue in https://github.com/directus/directus/issues/10842 👍🏻 

> At present knex is at 1.0.7. It seems a long way to get to v.2

The version number doesn't necessarily indicate any timing 🙂
